### PR TITLE
Feature - pull mainnet json

### PR DESCRIPTION
**Pull request by i25959341**

### CODE REVIEW

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+dsn = "postgresql://{}:{}@{}/{}".format(user, password, host, databasename)
+
+print(connection_str)
+
+def get_coz_mainnet_json():
+    r = requests.get(
+        'https://raw.githubusercontent.com/CityOfZion/neo-mon/master/docs/assets/mainnet.json')
+    return json.loads(r.text)
+
+def get_existed_nodes(cursor):
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: get_existing_nodes (typo)

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    nodes_dict = {}
+    cursor.execute("select id, hostname, ip from nodes")
+    results = cursor.fetchall()
+
+    for id, hostname, ip in results:
+        nodes_dict[hostname] = (id, hostname, ip)
+
+    return nodes_dict
+
+def get_existed_connections(cursor):
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: get_existing_connections

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                    "INSERT INTO nodes (hostname, ip) VALUES (%s, %s)", [hostname, ip])
+            else:
+                (id, hostnameFromDatabase, ipFromDatabase) = nodes_dict[hostname]
+
+                if ipFromDatabase!=ip:
+                    # IP has changes, some nodes uses loadbalancing and changes their ip all the time
+                    print("update node's ip. hostname: {} id: {} ipFromDatabase: {} ip:{}".format(hostname, id, ipFromDatabase, ip))
+                    cursor.execute("UPDATE nodes SET ip=%s WHERE id=%s;", [ip, id])
+            
+            # Update old connections table hack
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: Remove

File: neo-collector/insert/insertData.py [hunk 4]
```diff
@@ -40,12 +40,10 @@ def getSqlDateTime(ts):
 def getAddress(cursor):
     address_dict = {}
     reverse_address_dict = {}
-    cursor.execute("SELECT addr.id, concat(proto.protocol, '://', addr.address,':' , po.port) AS url \
```
Comment (f27d) [inline neo-collector/insert/insertData.py]: Change all nod to node, not sure why you would simplify here

File: neo-collector/insert/insertData.py [hunk 5]
```diff
@@ -57,38 +55,15 @@ def getAddress(cursor):
...
+            inner join nodes nod \
+            on nod.id=ce.node_id")
     ip_list = cursor.fetchall()
-    for source in ip_list:
-        ip_id, address_id, ip = source
+    for ip_id, address_id, ip in ip_list:
         address_dict[ip] = (ip_id, address_id, ip)
     return address_dict
 
-# def getAddressId(cursor):
```
Comment (f27d) [inline neo-collector/insert/insertData.py]: Remove comments like this, you can see them in GH history

File: neo-node/p2p_tcp/node.py [hunk 6]
```diff
@@ -28,7 +28,10 @@ def getSqlDateTime(ts):
 
 def getIpAddressMap(cursor):
     ip_dict = {}
-    cursor.execute("SELECT address_id, ip_tb.id AS ip_id, address, ip FROM address addr LEFT JOIN ip ip_tb ON addr.id=ip_tb.address_id")
+    cursor.execute("select ce.id, nod.id, nod.hostname, nod.ip \
```
Comment (f27d) [inline neo-node/p2p_tcp/node.py]: Switch to ``` ... ``` and then you can avoid the \

File: neo-redis/peerinfo/updateRedisPeers.py [hunk 7]
```diff
@@ -65,7 +65,7 @@
 
         r.set(redisNamespace+"edges", json.dumps(edges))
 
-        cursor.execute("select adr.id, address as hostname, protocol, CONCAT(proto.protocol, '://', adr.address) as address from address adr inner join protocol proto on adr.id = proto.address_id")
+        cursor.execute("select endpoint.id , nod.hostname, endpoint.protocol, CONCAT(endpoint.protocol, '://', nod.hostname) as address from connection_endpoints endpoint inner join nodes nod on nod.id = endpoint.node_id")
```
Comment (f27d) [inline neo-redis/peerinfo/updateRedisPeers.py]: nod -> Node

File: neo-collector/create/createOrUpdatePrimaryTables.py (hunk 1 above)
Comment (i25959341) [reply neo-collector/create/createOrUpdatePrimaryTables.py]: done

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+print(connection_str)
+
+def get_coz_mainnet_json():
+    r = requests.get(
+        'https://raw.githubusercontent.com/CityOfZion/neo-mon/master/docs/assets/mainnet.json')
+    return json.loads(r.text)
+
+def get_existing_nodes(cursor):
+    # For mainnet json, we use hostanme to check whether a node
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: Typo: hostname*

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+user = str(os.environ['PGUSER'])
+password = str(os.environ['PGPASSWORD'])
+
+connection_str = "dbname='{}' user='{}' host='{}' password='{}'".format(
+    databasename, user, host, password)
+dsn = "postgresql://{}:{}@{}/{}".format(user, password, host, databasename)
+
+print(connection_str)
+
+def get_coz_mainnet_json():
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: We need a bit more validation at this stage. Some simple checks to make sure the JSON file is in the format we expect

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+        
+if __name__ == "__main__":
+    while True:
+        connect_str = connection_str
+
+        conn = psycopg2.connect(connect_str)
+
+        cursor = conn.cursor()
+
+        data = get_coz_mainnet_json()
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: Need more checks here

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+        create_or_update_nodes_rows(cursor, data)
+        conn.commit()
+
+        create_connectionendpoints_rows(cursor, data)
+        conn.commit()
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: Do you need to commit twice?

File: neo-collector/create/createOrUpdatePrimaryTables.py [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    while True:
+        connect_str = connection_str
+
+        conn = psycopg2.connect(connect_str)
+
+        cursor = conn.cursor()
+
+        data = get_coz_mainnet_json()
+
+        if check_mainnet_json(data):
```
Comment (f27d) [inline neo-collector/create/createOrUpdatePrimaryTables.py]: What happens if it fails? Should add some logging/alerting.

**Pull request merged**